### PR TITLE
Multiple email recipients

### PR DESCRIPTION
**Pull request by bryankenote**
The goal of this was to allow alerts to be sent to multiple recipients. This change allows for a list of emails to be added to the config for each type of alert. `src/exampleConfig.ts` makes the intent fairly obvious.

### CODE REVIEW

File: src/types/IConfig.ts [hunk 1]
```diff
@@ -10,13 +11,8 @@ export interface IConfig {
...
 		ssl: boolean;
 	};
-	emailRecipient: {
-		text: string;
-		from: string;
-		to: string;
-		ssl: boolean;
-		port: number;
-	};
+	emailRecipients: { [alterType in AlertType]: string[] };
```
Comment (timothy-b) [inline src/types/IConfig.ts:14]:
```suggestion
	emailRecipients: { [alertType in AlertType]: string[] };

File: src/utilities/alerter.ts [hunk 2]
```diff
@@ -43,73 +56,38 @@ export function createAlerter(config: IConfig, log: Bunyan) {
 		const iAmHome = detectedMacs.has(config.myPortableDevice.mac);
 
 		if (iWasHome && !iAmHome) {
-			alertGoodbye();
+			triggerAlert('departure');
```
Comment (timothy-b) [inline src/utilities/alerter.ts:59]: Could you do something like
```suggestion
			triggerAlert(AlertType.departure);

File: src/utilities/alerter.ts [hunk 3]
```diff
@@ -4,14 +4,27 @@ import { IConfig } from '../types/IConfig';
...
-	text: string;
+const defaultAlertMessages = {
+	intruder:'The fortress is in peril.',
+	departure: 'Goodbye.',
+	arrival:'Welcome home.',
+	doorOpen:'The door is open.',
+	knownDevice:'A recognized device arrived.',
+}
+
+export type AlertType = keyof typeof defaultAlertMessages;
```
Comment (timothy-b) [inline src/utilities/alerter.ts:15]: 🤔 I feel like there should be a way in Typescript to match these keys to the corresponding keys in `emailRecipients` in `exampleConfig.ts`.

File: src/utilities/alerter.ts [hunk 4]
```diff
@@ -43,73 +56,38 @@ export function createAlerter(config: IConfig, log: Bunyan) {
...
-	}
+	function triggerAlert(type: AlertType, detectedKnownPortableDevices?: IDevice[]) {
+		const grammar = detectedKnownPortableDevices?.length > 1 ? 'are' : 'is';
 
-	async function alertNobodyHome() {
 		const email = {
-			...config.emailRecipient,
-			text: 'The fortress is in peril.',
+			...config.emailConfig,
+			to: config.emailRecipients[type].join(';'),
```
Comment (timothy-b) [inline src/utilities/alerter.ts:86]: Have you tested that delimiting the recipients with a semicolon works? I'd be happy to if not.

File: src/utilities/alerter.ts [hunk 5]
```diff
@@ -4,14 +4,27 @@ import { IConfig } from '../types/IConfig';
...
 import { arpscanDevicesAsync } from './scanner';
 
-interface IEmail {
-	text: string;
+const defaultAlertMessages = {
+	intruder:'The fortress is in peril.',
+	departure: 'Goodbye.',
+	arrival:'Welcome home.',
+	doorOpen:'The door is open.',
+	knownDevice:'A recognized device arrived.',
```
Comment (timothy-b) [inline src/utilities/alerter.ts:12]: nit: inconsistent spaces after colon. One space would be preferred,

File: src/utilities/alerter.ts (hunk 4 above)
Comment (bryankenote) [reply src/utilities/alerter.ts:80]: I have not. I have used email libraries that allowed this before. I think I remember being able to use `,` or `;`. Emailjs docs are useless though.

File: src/types/IConfig.ts [hunk 6]
```diff
@@ -10,13 +18,8 @@ export interface IConfig {
...
 		ssl: boolean;
 	};
-	emailRecipient: {
-		text: string;
-		from: string;
-		to: string;
-		ssl: boolean;
-		port: number;
-	};
+	emailRecipients: { [altertType in AlertType]: string[] };
```
Comment (timothy-b) [inline src/types/IConfig.ts:21]: still off lol
```suggestion
	emailRecipients: { [alertType in AlertType]: string[] };
```

File: src/utilities/alerter.ts [hunk 7]
```diff
@@ -1,15 +1,22 @@
...
-interface IEmail {
+const defaultAlertMessages: { [alertType in AlertType]: string } = {
+	intruder: 'The fortress is in peril.',
+	departure: 'Goodbye.',
+	arrival: 'Welcome home.',
+	doorOpen: 'The door is open.',
+	knownDevice: 'A recognized device arrived.',
+}
+
+const alertTypes = Object.keys(defaultAlertMessages).reduce((map, key) => ({ ...map, [key]: key }), {}) as { [alertType in AlertType]: AlertType };
```
Comment (timothy-b) [inline src/utilities/alerter.ts:15]: hah, nice 👍

Comment (bryankenote) [reply src/utilities/alerter.ts:15]: The type on `defaultAlertMessages` requires that all `AlertTypes` exist as keys. So `alertTypes` here should always contain all possible types. `function triggerAlert(type: AlertType, ...` would have caused an error for passing strings that were not valid alert types, but if this is prefered, it is just as valid.

**Pull request merged**